Managers/PresetMan.cpp

```diff
@@ -291,24 +294,77 @@ int PresetMan::GetModuleID(string moduleName)
     return -1;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// Method:          GetModuleNameFromPath
+//////////////////////////////////////////////////////////////////////////////////////////
+// Description:     Gets the Name of a loaded DataModule, from a full data file path.
+std::string PresetMan::GetModuleNameFromPath( std::string dataPath )
+{
+    if( dataPath.empty() ) {
+        return "";
+    }
+
+    int slashPos = dataPath.find_first_of( '/' );
+    if( slashPos == string::npos ) {
+        slashPos = dataPath.find_first_of( '\\' );
+    }
+    std::string moduleName = dataPath.substr( 0, slashPos );
+    // check if path starts with Data/ or Mods/ and remove that part to get to the actual module name
+    if( moduleName == "Data" || moduleName == "Mods" ) {
+        std::string shortenPath = dataPath.substr( slashPos + 1 );
+        slashPos = shortenPath.find_first_of( '/' );
+        if( slashPos == string::npos ) {
+            slashPos = shortenPath.find_first_of( '\\' );
+        }
+        moduleName = shortenPath.substr( 0, slashPos );
+    }
+
+    return moduleName;
+}
 
 //////////////////////////////////////////////////////////////////////////////////////////
 // Method:          GetModuleIDFromPath
 //////////////////////////////////////////////////////////////////////////////////////////
 // Description:     Gets the ID of a loaded DataModule, from a full data file path.
 
-int PresetMan::GetModuleIDFromPath(std::string dataPath)
+int PresetMan::GetModuleIDFromPath( std::string dataPath )
 {
-    if (dataPath.empty())
+    if( dataPath.empty() ) {
```

Comment:
```suggestion
    if (dataPath.empty()) {
```
In case this wasn't intentional, cause `if(` looks odd to me but `if ( dataPath.empty() )` would look fine as well